substrate/frame/examples/tasks-example/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use core::marker::PhantomData;
+
+use codec::{Decode, Encode};
+use frame_support::dispatch::DispatchResult;
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+use sp_runtime::DispatchError;
+
+#[frame_support::pallet(dev_mode)]
+pub mod pallet {
+
+	use super::*;
+	use frame_support::{pallet_prelude::*, traits::AggregatedTask};
+	use frame_system::pallet_prelude::*;
+
+	// this can be auto-generated from the macros
+	#[derive(Clone, PartialEq, Eq, Encode, Decode, TypeInfo)]
+	pub enum Task<T: Config> {
+		Increment,
+		Decrement,
+		#[doc(hidden)]
+		#[codec(skip)]
+		__Ignore(PhantomData<T>, frame_support::Never),
+	}
+
+	// this can be auto-generated from the macros and will always be the same
+	impl<T: Config> core::fmt::Debug for Task<T> {
+		fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+			f.debug_struct("Task").field("value", self).finish()
+		}
+	}
+
+	// we can automatically inject `InvalidTask` into an existing Error enum by finding it via
+	// visitor pattern, otherwise we can just emit an error enum containing just our
+	// `InvalidTask` variant. Alternatively we could just expect that `InvalidTask` is included
+	// in the error enum, by convention, or we could use something like
+	// `InvalidTransaction::Custom(1u8)` but this seems bad to me.
+	#[pallet::error]
+	pub enum Error<T> {
+		InvalidTask,
+		ValueOverflow,
+		ValueUnderflow,
+	}
+
+	// this will be auto-generated from `#[pallet::tasks]`
+	impl<T: Config> frame_support::traits::Task for Task<T>
+	where
+		T: TypeInfo,
+	{
+		type Enumeration = sp_std::vec::IntoIter<Task<T>>;
```

Comment:
Does this type need to be configurable? won't it always be `Vec`? 

Would be useful if you provide a fake example of the syntax that the end user will put under `#[pallet::task]` to clarify this.